src/main/java/thePackmaster/patches/energyandechopack/OnCreateCardPatch.java

```diff
@@ -0,0 +1,79 @@
+package thePackmaster.patches.energyandechopack;
+
+import com.evacipated.cardcrawl.modthespire.lib.ByRef;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch;
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInDiscardAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInDrawPileAction;
+import com.megacrit.cardcrawl.actions.common.MakeTempCardInHandAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.blue.DoubleEnergy;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import thePackmaster.packs.EnergyAndEchoPack;
+import thePackmaster.powers.energyandechopack.DivineTouchPower;
+import thePackmaster.powers.energyandechopack.PanacherPower;
+
+import static thePackmaster.util.Wiz.adp;
+
+public class OnCreateCardPatch {
```

Comment:
This works for most things, but it will miss a few cases because some cards use the effects like `ShowCardAndAddToHandEffect` directly instead of going through one of the `MakeTempCard` actions. For example, the `FlexibleDiscoveryAction` used by `DestinyDraw` uses `ShowCardAndAddToHandEffect`.

To catch everything reliably, I think you can adapt the patch at https://github.com/Jedi515/TheRanger/blob/master/src/TheRanger/patches/TempCardPatches.java, which patches `ShowCardAndAddToDiscardEffect`, `ShowCardAndAddToHandEffect`, and `ShowCardAndAddToDrawPileEffect`. It was written a while back but I still think it's the way to do this (I haven't tried that patch out myself but the idea behind it makes sense to me).